sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -20,12 +20,86 @@ package org.apache.spark.sql.execution.streaming.state
 import java.io._
 import java.nio.charset.Charset
 
+import scala.language.implicitConversions
+
 import org.apache.commons.io.FileUtils
+import org.apache.hadoop.conf.Configuration
 
 import org.apache.spark._
+import org.apache.spark.sql.catalyst.util.quietly
+import org.apache.spark.sql.execution.streaming.CreateAtomicTestManager
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.util.Utils
 
 class RocksDBSuite extends SparkFunSuite {
 
+  test("RocksDBFileManager: upload only new immutable files") {
+    val dfsRootDir = Utils.createTempDir().getAbsolutePath
+    val fileManager = new RocksDBFileManager(
+      dfsRootDir, Utils.createTempDir(), new Configuration)
+    val sstDir = s"$dfsRootDir/SSTs"
+    def numRemoteSSTFiles: Int = listFiles(sstDir).length
+    val logDir = s"$dfsRootDir/logs"
+    def numRemoteLogFiles: Int = listFiles(logDir).length
```

Comment:
Shall we have some code to check other files & metadata as well? We don't seem to cover the test on existence of zip file and the content.

Reply:
We'll enhance this test in the next PR, `RocksDBFileManager - load checkpoint from DFS`, by load back the metadata and check the other files. Let me keep this conversation open and be referenced in the next PR.

Reply:
Referenced this comment to https://github.com/apache/spark/pull/32767/files#diff-dc6f9dfe11e76f890ff2986f866853bcac263027c82562f9a52f4672a5460826R97